pkg/messages/errorcodes/errorcodes.go

```diff
@@ -0,0 +1,103 @@
+	ConfigurationStoreNotConfigured = "ERR_CONFIGURATION_STORE_NOT_CONFIGURED"
+	ConfigurationStoreNotFound      = "ERR_CONFIGURATION_STORE_NOT_FOUND"
+	ConfigurationSubscribe          = "ERR_CONFIGURATION_SUBSCRIBE"
+	ConfigurationUnsubscribe        = "ERR_CONFIGURATION_UNSUBSCRIBE"
+	ConversationInvalidParms        = "ERR_CONVERSATION_INVALID_PARMS"
+	ConversationInvoke              = "ERR_CONVERSATION_INVOKE"
+	ConversationMissingInputs       = "ERR_CONVERSATION_MISSING_INPUTS"
+	ConversationNotFound            = "ERR_CONVERSATION_NOT_FOUND"
+	Crypto                          = "ERR_CRYPTO"
+	CryptoKey                       = "ERR_CRYPTO_KEY"
+	CryptoProviderNotFound          = "ERR_CRYPTO_PROVIDER_NOT_FOUND"
+	CryptoProvidersNotConfigured    = "ERR_CRYPTO_PROVIDERS_NOT_CONFIGURED"
+	DirectInvoke                    = "ERR_DIRECT_INVOKE"
+	GetWorkflow                     = "ERR_GET_WORKFLOW"
+	HealthAppidNotMatch             = "ERR_HEALTH_APPID_NOT_MATCH"
+	HealthNotReady                  = "ERR_HEALTH_NOT_READY"
+	IllegalKey                      = "ERR_ILLEGAL_KEY"
+	InstanceIdInvalid               = "ERR_INSTANCE_ID_INVALID"
+	InstanceIdNotFound              = "ERR_INSTANCE_ID_NOT_FOUND"
+	InstanceIdProvidedMissing       = "ERR_INSTANCE_ID_PROVIDED_MISSING"
+	InstanceIdTooLong               = "ERR_INSTANCE_ID_TOO_LONG"
+	Internal                        = "ERR_INTERNAL"
+	InvokeOutputBinding             = "ERR_INVOKE_OUTPUT_BINDING"
+	LockStoreNotConfigured          = "ERR_LOCK_STORE_NOT_CONFIGURED"
+	LockStoreNotFound               = "ERR_LOCK_STORE_NOT_FOUND"
+	MalformedRequest                = "ERR_MALFORMED_REQUEST"
+	MalformedRequestData            = "ERR_MALFORMED_REQUEST_DATA"
+	MalformedResponse               = "ERR_MALFORMED_RESPONSE"
+	NotConfigured                   = "ERR_NOT_CONFIGURED"
+	NotFound                        = "ERR_NOT_FOUND"
+	NotSupported                    = "ERR_NOT_SUPPORTED"
+	NotSupportedStateOperation      = "ERR_NOT_SUPPORTED_STATE_OPERATION"
+	OutboundHealthNotReady          = "ERR_OUTBOUND_HEALTH_NOT_READY"
+	PauseWorkflow                   = "ERR_PAUSE_WORKFLOW"
+	PermissionDenied                = "ERR_PERMISSION_DENIED"
+	PostfixQueryFailed              = "ERR_POSTFIX_QUERY_FAILED"
+	PublishOutbox                   = "ERR_PUBLISH_OUTBOX"
+	PubsubCloudEventsSer            = "ERR_PUBSUB_CLOUD_EVENTS_SER"
+	PubsubEmpty                     = "ERR_PUBSUB_EMPTY"
+	PubsubEventsSer                 = "ERR_PUBSUB_EVENTS_SER"
+	PubsubForbidden                 = "ERR_PUBSUB_FORBIDDEN"
+	PubsubNotConfigured             = "ERR_PUBSUB_NOT_CONFIGURED"
+	PubsubNotFound                  = "ERR_PUBSUB_NOT_FOUND"
+	PubsubPublishMessage            = "ERR_PUBSUB_PUBLISH_MESSAGE"
+	PubsubRequestMetadata           = "ERR_PUBSUB_REQUEST_METADATA"
+	PurgeWorkflow                   = "ERR_PURGE_WORKFLOW"
+	RaiseEventWorkflow              = "ERR_RAISE_EVENT_WORKFLOW"
+	ResumeWorkflow                  = "ERR_RESUME_WORKFLOW"
```

Comment:
This is a nit, but can you make all the API specific error codes start with the respective API? So for ex: 
```suggestion
	WorkflowResume                  = "ERR_RESUME_WORKFLOW"
```
This will help developers later on if we standardize this a bit more by putting the API first.  I see several in this list: `PauseWorkflow`, `GetWorkflow`, `PurgeWorkflow`, `RaiseEventWorkflow`, etc

Then can you also group the errorcodes in this file by API? I see a group of workflow API error codes below, so this should get moved down there

Reply:
+1 on this